subsys/bluetooth/controller/hci/hci.c

```diff
@@ -6886,6 +6884,14 @@ static void le_ext_adv_report(struct pdu_data *pdu_data,
 				return;
 			}
 
+			if (node_rx_curr == node_rx && !node_rx_next) {
+				/* BT Core 5.4, Vol 6, Part B, Section 4.4.3.5:
+				 * If the Controller does not listen for or does not receive the
+				 * AUX_ADV_IND PDU, no report shall be generated
+				 */
+				return;
+			}
```

Comment:
Dont we need a call to `node_rx_extra_list_release()`? Can we combine with previous `return` on PHY check?

Reply:
> Dont we need a call to `node_rx_extra_list_release()`?

In this case there are no extra nodes to release, so the call would just be `node_rx_extra_list_release(NULL)` - which doesn't do anything.

> Can we combine with previous `return` on PHY check?

Yes I can do that - I'll need to work on getting an understandable comment though.